torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -120,17 +120,19 @@ std::tuple<std::vector<at::Tensor>, c10::intrusive_ptr<Work>> reduce_scatter_(
       output_tensors, work);
 }
 
-c10::intrusive_ptr<Work> _reduce_scatter_base_(
+std::tuple<at::Tensor, c10::intrusive_ptr<Work>> _reduce_scatter_base_(
```

Comment:
curious, when will this method be invoked and when will we dispatch to the cpu/cuda version below?

Reply:
Yes, the definition in Ops.cpp is redundant and will get removed in https://github.com/pytorch/pytorch/pull/88330, feel free to land this PR as-is though!